proposal.rst

```diff
@@ -262,7 +276,8 @@ and take huge space.
 **Please note any uncertainties or aspects which depend on further research or
 investigation.**
 
-I am still not sure if we can come up with better approach. Or how better the result would be for approach using the first and last doc ids
+We still assume independance in calculating the estimates.
```

Comment:
Well, yes and no.  Currently we always just assume the terms occur independently.

Using the docid range information effectively feeds some information about dependencies between terms into the calculation.  In some cases that gives us perfect information (for example, in the degenerate case where the terms have disjoint ranges of docids we can tell that they don't occur together) whereas in others we get no useful information and have to assume independence (for example, if the terms both occur in the first and in the last used docids we'd just use the same estimate as currently).

So we do still assume independence in some cases, but that's an improvement on always assuming it.